src/backend/hip/subconfigure.m4

```diff
@@ -11,38 +11,23 @@
 # --with-hip-sm
 AC_ARG_WITH([hip-sm],
             [
-  --with-hip-sm=<options> (https://reviews.llvm.org/differential/changeset/?ref=2126544)
+  --with-hip-sm=<options>
           Comma-separated list of below options:
+                auto - letting HIPCC to auto detect target_id
```

Comment:
auto-detection is a great additional. the "all" target took forever to build in my experience.
```suggestion
                auto - let HIPCC auto detect the target_id
```

Reply:
Yanfei and I had discussed offline for a bit. I think we can simply remove the "all" and the subsets options. By default, it is essentially "auto", where `hipcc` do whatever its default. Then user always can pass in specific list of sm, which we directly pass on to `hipcc`. I don't think it is worth while for us to maintain a separate option behavior from `hipcc`'s.